proposals/1/standardized-mod-format.md

```diff
@@ -0,0 +1,126 @@
+# Standardized Mod Format (DRAFT!)
+
+## Description Of Issue
+
+Lack of a clear specification between modloaders on a common supported format.
+
+## Solution
+
+A specification which exactly defines specific terms, and does not define legacy formats from the previous era.
+
+### Advantages
+
+- Modloaders and modders know what is supported everywhere.
+
+- Less likelihood to use outdated mechanisms from the old era
+
+- APIs defined in other specifications may gain reference implementations using this specification - these can then become the only implementation of those APIs, preventing fragmentation.
+
+### Disadvantages
+
+- Per-modloader extensions not covered, theoretically limiting features for more portable mods.
+
+### Regarding Implementation
+
+It is important to note that some features of this specification may, themselves, be implemented within mods.
+
+This is fine - the implementation of this specification is then a matter of "that modloader + these mods".
+
+This in particular applies to how CCLoader handles patch files via `simplify`.
+
+## Mod Format
+
+A mod is in a directory. The directory's name is the ID of the mod.
+
+The directory then contains a file called "package.json".
+
+### The package.json format
+
+The package.json format describes a mod.
+
+It has the following elements:
+
+```
+{
+ "version": <Semver-compliant version string>,
+ "ccmodDependencies": <Optional Object - keys are mod IDs, values are semver-compliant version constraints - This is used between different mods>,
+ "dependencies": <
+  Optional Object - due to compatibility, this is a bit awkward
+  If ccmodDependencies is provided, keys are Node modules, values are semver-compliant version constraints
+  If ccmodDependencies is NOT provided, this key is effectively renamed to it - meaning keys are mod IDs, not Node modules.
+ >,
+
+ "name": <Optional human-readable string - otherwise, the directory name is used>,
+ "description": <Optional human-readable string>,
+
+ "preload": <Optional filename of a script>,
+ "postload": <Optional filename of a script>,
+ "prestart": <Optional filename of a script>,
+ "main": <Optional filename of a script. For example, "mod.js" loads the file "mod.js" within the mod folder>,
+
+ "usesRequire": <Optional boolean - if true, this mod uses the 'require' function. Defaults to true>,
+
+ "assets": <
+  Optional array of strings, such as "media/icons.png".
+  An asset ending in ".json.patch" is a patch file - see Patch Files.
+  If not provided, assets may be automatically guessed or completely ignored.
```

Comment:
Keeping this up to date is a real pain. Should this be really ignored if missing? Maybe add a `true` or dirname option to automatically infer assets.

Reply:
Impossible if non-local.
If the specification should or should not support non-local is up for debate, though.